### DISCUSSION TITLE
Does the order of the `interactions` matter?

### DISCUSSION THREAD

**Discussion by r100-stack** [floating-ui/floating-ui#2936] [Q&A]
Related: [#1893](https://github.com/floating-ui/floating-ui/issues/1893)

We are hoping to create a menu with the click and hover `interactions` on the trigger. One additional requirement is that clicking the trigger repeatedly should toggle the trigger.

I did this by first adding the following interactions (attempt 1 from the playground):

```jsx
[
  useHover(context, { move: false }),
  useClick(context, { toggle: true }),
]
```

This caused the need for users to click the trigger *twice* to dismiss the menu for the first time (similar to [#1893](https://github.com/floating-ui/floating-ui/issues/1893)). To avoid this, I tried adding `useDismiss` as suggested in that issue (attempt 2.1 from the playground):

```jsx
[
  useHover(context, { move: false }),
  useClick(context, { toggle: false }),
  useDismiss(context, {
    referencePress: true,
    referencePressEvent: "click",
  }),
]
```

But this didn't toggle the menu upon clicks as we wanted. Clicking simply dismissed the menu all the time. So, my teammate and I thought to try re-ordering the triggers (attempt 2.2 from the playground):

```jsx
[
  useHover(context, { move: false }),
  useDismiss(context, {
    referencePress: true,
    referencePressEvent: "click",
  }),
  useClick(context, { toggle: false }),
]
```

This re-order now toggled the menu correctly upon click, without even needing users to click the trigger twice to dismiss menu for the first time 🎉. So, we thought that order of the interactions matters even thought we could not find a mention of it in the [`useInteraction`](https://floating-ui.com/docs/useInteractions) docs.

But updating the sandbox from `0.24.0` to `latest` (`0.26.17`) broke it again.

So, to confirm, does the order of the interactions have any effect?

Playgrounds:
* [`0.24.0`](https://codesandbox.io/p/sandbox/relaxed-dhawan-575pdw?file=%2Fsrc%2FApp.tsx)
* [`latest` (`0.26.17`)](https://codesandbox.io/p/sandbox/blissful-monad-ffpvnq?file=%2Fsrc%2FApp.tsx)

**Comment by atomiks** [comment]
In some rare cases, yes it does matter. This is not well-documented though right now.
